packages/design-system-react/src/components/text/README.mdx

```diff
@@ -185,30 +185,49 @@ Use the boolean `ellipsis` prop to make the `Text` component have an ellipsis.
 
 <Canvas of={TextStories.Ellipsis} />
 
-### As
+### AsChild
 
-Use the `as` prop to change the root HTML element of the `Text` component.
+Use the `asChild` prop to compose the Text component with other components while maintaining the Text's styles. This allows you to merge the Text component's styles onto its child component instead of rendering an additional DOM element.
 
-You can also utilize the `ValidTag` enum from `@metamask/design-system-react` to ensure that you are using a valid HTML element.
+When using `asChild`, you must provide exactly one child element that can accept a className prop. This works with both interactive elements like buttons and inputs, as well as standard HTML elements.
 
-<Canvas of={TextStories.As} />
+<Canvas of={TextStories.AsChild} />
 
-Renders the HTML:
+Examples:
 
-```html
-<span>span</span> <strong>strong</strong>
-```
+```jsx
+// Span example
+<Text asChild className="block">
+  <span>Text rendered as span</span>
+</Text>
 
-### Strong
```

Comment:
No longer supported we can use the `font-bold` classname or ideally the `asChild` prop.